static/src/javascripts/projects/common/modules/spacefinder.js

```diff
@@ -111,7 +111,7 @@ const onInteractivesLoaded = memoize((rules) => {
 const filter = (list, filterElement, exclusions) => {
 	const filtered = [];
 	list.forEach((element) => {
-		if (filterElement(element)) {
+		if (filterElement(element, filtered.at(-1))) {
```

Comment:
Wow Spacefinder code is particularly hard to read!

Things that are not so nice here:
- mutating an array and also using the same array to compare the last element (a result of the original code not the fix)
- it is mutating one of the parameters passed into it as well as returning a result
- it is hard to read and reason about

This is probably a refactor outside of this bugfix and we'll need some safety in better unit tests but maybe a small improvement is that this function can become pure i.e. rather than mutating the parameter, it returns an array or object of `exclusions` and `filtered`

Reply:
Hey @arelra, thanks for the review! I like your suggestion and have made the change in a separate commit:

https://github.com/guardian/frontend/pull/24592/commits/73b39dd027e279dfd22b9388aaba1e1cc1b1fc5a

I'm undecided whether it belongs here or in a separate PR to keep the intention of this PR nice and focussed.

Reply:
Nice! I personally think the refactor commit is okay for this PR, we often tidy things up as we touch them.

But I agree it's probably wise to postpone any bigger refactors to another PR.